### PR TITLE
implement garabage collection

### PR DESCRIPTION
**Pull request by jrushlow**
- adds ability to remove expired reset request objects
- helper calls garbage collection on every create & validation call
- enabled by default, can be disabled by setting `enable_garbage_collection` to false
- provides command to manually force garbage collection

closes tkt #8

### CODE REVIEW

File: src/Command/CleanerCommand.php [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+use Symfony\Component\Console\Output\OutputInterface;
+use SymfonyCasts\Bundle\ResetPassword\Util\ResetPasswordCleaner;
+
+/**
+ * @author  Jesse Rushlow <jr@rushlow.dev>
+ * @author Ryan Weaver <ryan@symfonycasts.com>
+ *
+ * @final
+ */
+class CleanerCommand extends Command
```
Comment (weaverryan) [inline src/Command/CleanerCommand.php]: ResetPasswordRemoveExpiredCommand - it's always nice to name it close to the command name... it helps when I try to search for a command class lazily based on its command string :)

File: src/Command/CleanerCommand.php [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+     * @inheritDoc
+     * @psalm-suppress InvalidReturnType
+     */
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $output->writeln('Removing expired reset password requests...');
+
+        $intRemoved = $this->cleaner->handleGarbageCollection(true);
+
+        $output->writeln('Garbage collection successful. Removed '.$intRemoved.' reset password request objects.');
```
Comment (weaverryan) [inline src/Command/CleanerCommand.php]: Use `sprintf()` here - it's just how it's always done in Symfony

File: src/Persistence/ResetPasswordRequestRepositoryInterface.php [hunk 3]
```diff
@@ -21,4 +21,9 @@ public function findResetPasswordRequest(string $selector): ?ResetPasswordReques
     public function getMostRecentNonExpiredRequestDate(object $user): ?\DateTimeInterface;
 
     public function removeResetPasswordRequest(ResetPasswordRequestInterface $resetPasswordRequest): void;
+
+    /**
+     * Remove all expired reset password request objects from persistence
```
Comment (weaverryan) [inline src/Persistence/ResetPasswordRequestRepositoryInterface.php]: Needs an `@return` to describe the meaning of the `int` return type

File: src/Util/ResetPasswordCleaner.php [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    /**
+     * Clears expired reset password requests from persistence.
+     *
+     * Enable/disable in configuration. Calling with $force = true
+     * will attempt to remove expired requests regardless of
+     * configuration setting.
+     */
+    public function handleGarbageCollection(bool $force = false): int
+    {
+        if ($this->enabled || (!$this->enabled && $force)) {
```
Comment (weaverryan) [inline src/Util/ResetPasswordCleaner.php]: Can be simplified:

```suggestion
        if ($this->enabled || $force) {
```

File: src/tests/IntegrationTests/Command/CleanerCommandTest.php [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+/**
+ * @author Jesse Rushlow <jr@rushlow.dev>
+ * @author Ryan Weaver   <ryan@symfonycasts.com>
+ */
+class CleanerCommandTest extends KernelTestCase
+{
+    public function testGarbageCollection(): void
+    {
+        //@TODO Remove skip after setting up integration tests
+        $this->markTestIncomplete('IntegrationTests not setup yet.');
```
Comment (weaverryan) [inline src/tests/IntegrationTests/Command/CleanerCommandTest.php]: Do you still want to add an integration test?

File: src/Command/CleanerCommand.php (hunk 1 above)
Comment (jrushlow) [reply src/Command/CleanerCommand.php]: done

File: src/Command/CleanerCommand.php (hunk 2 above)
Comment (jrushlow) [reply src/Command/CleanerCommand.php]: done

File: src/Persistence/ResetPasswordRequestRepositoryInterface.php (hunk 3 above)
Comment (jrushlow) [reply src/Persistence/ResetPasswordRequestRepositoryInterface.php]: done

File: src/Util/ResetPasswordCleaner.php (hunk 4 above)
Comment (jrushlow) [reply src/Util/ResetPasswordCleaner.php]: done

File: src/tests/IntegrationTests/Command/CleanerCommandTest.php (hunk 5 above)
Comment (jrushlow) [reply src/tests/IntegrationTests/Command/CleanerCommandTest.php]: yes but ready to do so now.. we need either functional or integration tests (possibly both) to ensure our configuration is working as intended.. can discuss that later..

removed the test for now, will implement it in the future..

Comment (weaverryan) [reply src/tests/IntegrationTests/Command/CleanerCommandTest.php]: Awesome - keep it on the list - I also like to have at least one integration test that shows that the container will boot up correctly with the bundle installed & that any service with autowireable aliases can be fetched & instantiated via autowiring - like this here https://github.com/symfony/webpack-encore-bundle/blob/5c0f659eceae87271cce54bbdfb05ed8ec9007bd/tests/IntegrationTest.php#L143-L149

**Pull request merged**